### PR TITLE
Try adding windows workflow

### PR DESCRIPTION
**Pull request by jemmaissroff**
just for looking at CI

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -20,6 +20,11 @@ jobs:
       with:
         ruby-version: head
         bundler-cache: true
+    - name: Install MinGW (Windows)
```
Comment (flavorjones) [inline .github/workflows/main.yml:23]: I see this is a draft but I wanted to note that I think setup-ruby comes with mingw installed on Windows. This step might not be necessary?

Comment (eregon) [reply .github/workflows/main.yml:23]: Yes, it shouldn't be

**Comment by eregon** [comment]
https://github.com/ruby/yarp/actions/runs/5225354819/jobs/9434710232?pr=966#step:6:26
`cc ... -o build/librubyparser.so` so that needs to be fixed most likely, I think it should be `rubyparser.dll` or so

**Comment by jemmaissroff** [comment]
> I think it should be `rubyparser.dll` or so

@eregon do you mind explaining why?

**Comment by eregon** [comment]
On Windows SOEXT is I think `dll` and not `so`
Also the "libprefix" on Windows is "", vs "lib" for most other OS.
@MSP-Greg you know a lot more about this, maybe you could advice?

**Comment by MSP-Greg** [comment]
If the file is loaded in Ruby via `require`, it would normally have an `.so` extension.  If it's loaded by fiddle/ffi, it would have a `.dll` extension.

I can't do so right now, but later I'll look at this.

**Comment by MSP-Greg** [comment]
Re Windows:

I changed `Makefile` to start with the following:
```make
ifeq ($(shell uname), Darwin)
SOEXT := dylib
else ifeq ($(shell uname), Linux)
SOEXT := so
else
SOEXT := dll
endif
```

In `yarp/.rb`, loaded `librubyparser.dll` with Fiddle.

Running `parser_test.rb` causes a SEGV, but all other tests pass.

**Comment by jemmaissroff** [comment]
Punting this for later work! Created #980

**Pull request closed without merging**